### PR TITLE
【paddle.fleet】add localsgd meta optimizer

### PR DESCRIPTION
**Pull request by gavin1332**
### PR types
New features

### PR changes
APIs

### Describe
add localsgd meta optimizer. Here is an example,
```
import paddle.fleet as fleet
from paddle.fleet import PaddleCloudRoleMaker

role = PaddleCloudRoleMaker(is_collective=True)
fleet.init(role)

input_x = paddle.fluid.layers.data(name="x", shape=[32], dtype='float32')
input_y = paddle.fluid.layers.data(name="y", shape=[1], dtype='int64')

fc = paddle.fluid.layers.fc(input=input_x, size=64, act='tanh')
prediction = paddle.fluid.layers.fc(input=[fc], size=2, act='softmax')
cost = paddle.fluid.layers.cross_entropy(input=prediction, label=input_y)
avg_cost = paddle.fluid.layers.mean(x=cost)

strategy = paddle.fleet.DistributedStrategy()
strategy.localsgd = True
strategy.auto = True
config = strategy.localsgd_configs
config['k_steps'] = 1 
strategy.localsgd_configs = config

optimizer = paddle.optimizer.SGD(learning_rate=0.01)
optimizer = fleet.distributed_optimizer(optimizer, strategy=strategy)
optimizer.minimize(avg_cost)
```

### CODE REVIEW

File: python/paddle/fleet/meta_optimizers/collective_helper.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                'rank': rank,
+                'ring_id': ring_id,
+                self.op_role_key: OpRole.Forward
+            })
+
+    def _broadcast_params(self):
+        block = self.startup_program.global_block()
+        ring_id = -1
+        for param in block.iter_parameters():
+            if param.is_distributed:
```
Comment (sandyhouse) [inline python/paddle/fleet/meta_optimizers/collective_helper.py]: In some cases (e.g., when using GradientClipByGlobalNorm), the type of param in the pair of param_grad is Variable which has no attribute is_distributed. How to deal with these cases?

Comment (gavin1332) [reply python/paddle/fleet/meta_optimizers/collective_helper.py]: Thanks for the advice. However, I have not found the code about replacement of `Parameter` to `Variable` in `param_grad`.
the source code I referenced is: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/clip.py

Comment (sandyhouse) [reply python/paddle/fleet/meta_optimizers/collective_helper.py]: 是的。使用GradientClipByGlobalNorm不会导致类型从Parameter变为Variable，请忽略我上面的注释。和用户一起定位到出现下午提到的程序在此处代码报错的问题，是由于用户的一些比较特殊的操作引起的。

**Review by guru4elephant — APPROVED** [review]
LGTM

File: python/paddle/fleet/meta_optimizers/collective_helper.py (hunk 1 above)
Comment (gavin1332) [reply python/paddle/fleet/meta_optimizers/collective_helper.py]: Get it. As `Parameter` has a specific definition, I think it is a bad idea for any user to replace `Parameter` to `Variable` in `param_grad` structure.

**Pull request merged**